src/ChainSafe.Gaming/Web3/Evm/Wallet/WalletTransactionExecutor.cs

```diff
@@ -27,10 +28,20 @@ public WalletTransactionExecutor(IWalletProvider walletProvider, ISigner signer)
             this.signer = signer;
         }
 
-        public async Task<TransactionResponse> SendTransaction(TransactionRequest transaction)
+        public async Task<TransactionResponse> SendTransaction(TransactionRequest transaction, IGasFeeModifier gasFeeModifier = null)
         {
             transaction.From ??= signer.PublicAddress;
             transaction.Data ??= "0x";
+            gasFeeModifier ??= new NoGasFeeModifier();
+
+            if (!transaction.GasPrice.Value.IsZero)
+            {
+                transaction.GasPrice = gasFeeModifier.ModifyGasFee(transaction.GasPrice);
+            }
+            else if(!transaction.MaxFeePerGas.Value.IsZero)
+            {
+                transaction.GasPrice = gasFeeModifier.ModifyGasFee(transaction.GasPrice);
```

Comment:
this should probably use `transaction.MaxFeePerGas` not `transaction.GasPrice`, right?